vowpalwabbit/core/include/vw/core/parse_args.h

```diff
@@ -49,6 +50,7 @@ inline bool ends_with(const std::string& full_string, const std::string& ending)
   return VW::ends_with(full_string, ending);
 }
 
+void handling_separator(VW::workspace& all, std::string& str, const std::string& name, const char default_symbol);
```

Comment:
Could you explain what this function is for?

Reply:
This function is for checking the input of the separator. Since we handle the namespace and field separator pretty much the same way, I make that handling and checking process into a function.

This function first check if the separator starts with `\t`. Since pressing Tab has their special meaning in CmdLine (AutoComplete). You can't type the tab exactly in CmdLine. In such a situation, users can just type `\t` as the separator to represent tabs. In addition, otherwise if assigning more than one character, a warning will be thrown, and will ignore the other characters and only accept the initial character. If no separator is specified, will return to default.

Reply:
Thanks for explaining!